compiler/rustc_parse_format/src/lib.rs

```diff
@@ -939,6 +954,17 @@ fn find_width_map_from_snippet(
     (width_mappings, true)
 }
 
+fn unescape_string(string: &str) -> string::String {
+    let mut buf = string::String::new();
+    unescape::unescape_literal(string, unescape::Mode::Str, &mut |_, unescaped_char| {
+        match unescaped_char {
+            Ok(c) => buf.push(c),
+            Err(_) => {}
```

Comment:
Hm, should we handle escape errors and bail in `find_width_map_from_snippet` if we see any?

Reply:
I dont think it matters much but doesnt hurt